app/src/main/java/com/github/polypoly/app/map/MapActivity.kt

```diff
@@ -136,6 +144,81 @@ class MapActivity : ComponentActivity() {
         }, modifier = Modifier.testTag("map"))
     }
 
+    /**
+     * Button for rolling the dice.
+     */
+    @Composable
+    fun RollDiceButton() {
+        Box(modifier = Modifier.fillMaxWidth()) {
+            Button(
+                modifier = Modifier
+                    .size(80.dp)
+                    .align(Alignment.BottomCenter)
+                    .offset(y = (-80).dp)
+                    .testTag("rollDiceButton"),
+                onClick = {
+                    // Toast.makeText(this@MapActivity, rollDice(), Toast.LENGTH_SHORT).show()
+                    showRollDiceDialog.value = true
+                },
+                shape = CircleShape
+
+            ) {
+                Icon(Icons.Filled.Casino, contentDescription = "Roll Dice")
+            }
+        }
+    }
+
+    /**
+     * Dice roll dialog, shows the result of 3 dice rolls in a column.
+     */
+    @Composable
+    fun RollDiceDialog() {
+        if (showRollDiceDialog.value) {
+            Dialog(onDismissRequest = { showRollDiceDialog.value = false }) {
+                AlertDialog(
+                    onDismissRequest = { showRollDiceDialog.value = false },
+                    title = { Text("Dice Roll") },
+                    text = {
+                        Column {
+                            val rollDice = rollDiceLocations()
+                            // 3 buttons, containing the name of a random location
+                            for (i in 0..2)
+                                Button(onClick = { }) {
+                                    Text(rollDice[i].name)
+                                }
+                        }
+                    },
+                    confirmButton = {
+                        Button(onClick = { showRollDiceDialog.value = false }) {
+                            Text("Quit")
+                        }
+                    }
+                )
+            }
+        }
+    }
+
+    /**
+     * Rolls the dice and returns the location that corresponds to the sum of 2 dice rolls, 3 times
+     * ensuring that the player does not visit the same location twice.
+     */
+    private fun rollDiceLocations(): List<com.github.polypoly.app.game.Location> {
+        val locationsNotToVisitName = mutableListOf(mapViewModel.closeLocation.value?.name)
+
+        val locationsToVisit = mutableListOf<com.github.polypoly.app.game.Location>()
+        for (i in 1..3) {
+            val diceRollsSum = IntArray(2) { (1..6).random() }.sum() - 2
```

Comment:
This is an interesting choice. I see how it makes sense for fidelity to the original game, but it seems a bit confusing given that you have more dimensions than traditional Monopoly 🤔 looking forward to an explanation in the game rules